lasy/profiles/from_openpmd_profile.py

```diff
@@ -110,13 +116,23 @@ def __init__(
         if not is_envelope:
             grid = create_grid(F, axes, dim, is_envelope=is_envelope)
             grid, omg0 = field_to_envelope(grid, dim, phase_unwrap_nd)
-            array = grid.get_temporal_field()[0]
+            array = grid.get_temporal_field()
         else:
-            s = io.Series(path + "/" + prefix + "_%T.h5", io.Access.read_only)
+            if is_waket:
+                filepath = path + "/" + prefix + "%T.h5"
+            else:
+                filepath = path + "/" + prefix + "_%T.h5"
+            s = io.Series(filepath, io.Access.read_only)
             it = s.iterations[iteration]
-            omg0 = it.meshes["laserEnvelope"].get_attribute("angularFrequency")
+            omg0 = it.meshes[field].get_attribute("angularFrequency")
             array = F
 
+        if lambda0:
+            omg0 = 2 * np.pi * c / lambda0
+
+        if field == "a":
+            array = (m_e * c * omg0 / e) * array
```

Comment:
Could you use `vector_potential_to_field` instead?

Reply:
Perhaps we could, but what is `vector_potential_to_field` exactly doing here?
```
A =  -np.gradient(field, grid.axes[-1], axis=-1, edge_order=2) + 1j * omega0 * field
a = m_e * c / e * A
```

Reply:
This is using the definition of vector potential, basically $E_x = -\frac{\partial A_x}{\partial t}$, with correction for the envelope. This makes sure all the terms including CEP are correct.